packages/playground/website/src/components/Toast/index.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import './css/toast.css';
+import useQuery from '../../hooks/useQuery';
+
+const Toast: React.FC = () => {
+	const onBoarding = useQuery('onBoarding');
+	const [showToast, setShowToast] = React.useState(false);
+	const TOAST_MESSAGE =
+		'All of your modification are private, and will be lost after a page refresh!';
+
+	const showToastMessage = (message: string): void => {
+		/**
+		 * @param message: string, options: object
+		 * @see https://fkhadra.github.io/react-toastify/introduction/
+		 */
+		toast.info(message, {
+			autoClose: 20000,
+			position: toast.POSITION.BOTTOM_CENTER,
+		});
+	};
+
+	React.useEffect(() => {
+		if (onBoarding && parseInt(onBoarding) === 1) {
+			// Get the isOnboarded flag value from browser's local storage.
+			const isOnboarded = window.localStorage.getItem('isOnboarded');
+
+			if (!isOnboarded) {
+				window.localStorage.setItem('isOnboarded', '1');
+				setShowToast(true);
+				showToastMessage(TOAST_MESSAGE);
+			} else {
+				setShowToast(false);
+			}
+		}
+	}, [onBoarding]);
+
+	return (
+		<div>{showToast && <ToastContainer className="toast-container" />}</div>
+	);
+};
+
+export default Toast;
```

Comment:
I like where you're going with this, but I think I overcommunicated how much I thought belonged in the separate component.

the Toast is unto itself a generic UI component. the onboarding message is a good candidate for something that _uses_ the toast system. image splitting these into separate components.

```jsx
// browser-chrome.tsx
…
<ToastContainer className="toast-container" />
```

```jsx
// onboarding-message.tsx
…

export const OnboardingMessage = () => {
	useEffect( () => {
		…

		if ( showOnboarding ) {
			toast.info( … );
		}
	}, [] );
}
```

We don't need to call `useState` to set the toast message, as it's inherently an imperative API. we can call that once in a `useEffect` with an empty array as its dependencies. Even though the query string in the URL _could_ change, I don't see a need to pop up the onboarding message if it were to dynamically add the expected parameter via JS.